File: praktikumsplaner-frontend/frontend/src/types/Studiengang.ts
export const Studiengang = [
    {
        name: "Wirtschaftsinformatik (BWI)",
        value: "BWI",
        color: "pink lighten-4",
    },
    { name: "Informatik (B.Sc.)", value: "BSC", color: "green lighten-3" },
    {
        name: "Verwaltungsinformatik (VI)",
        value: "VI",
        color: "deep-purple lighten-3",
    },
] as const;

export function findStudiengangColorByValue(value: string): string {
    const studiengang = Studiengang.find((s) => s.value === value);
    return studiengang ? studiengang.color : "";
}

File: praktikumsplaner-backend/src/main/java/de/muenchen/oss/praktikumsplaner/domain/enums/Studiengang.java
package de.muenchen.oss.praktikumsplaner.domain.enums;

import lombok.Getter;

@Getter
public enum Studiengang {
    BSC("Bachelor of Science - Informatik"),
    BWI("Wirtschaftsinformatik kommunal"),
    VI("Diplomverwaltungsinformatiker*in"),
    LLB("Bachelor of Laws"),
    PUMA("Public Management/Public Sector"),
    QE3("Diplomverwaltungswirt*in");

    private final String longName;

    Studiengang(final String longName) {
        this.longName = longName;
    }

    public static Studiengang fromBildungsrichtung(final Bildungsrichtung bildungsrichtung) {
        if (bildungsrichtung.getArt() != Bildungsrichtung.Art.STUDIUM) {
            throw new IllegalArgumentException("Bildungsrichtung " + bildungsrichtung + " ist kein Studiengang");
        }
        return Studiengang.valueOf(bildungsrichtung.name());
    }
}
